core/src/main/java/org/infinispan/stream/impl/DistributedCacheStream.java

```diff
@@ -537,23 +470,130 @@ public long count() {
+                  segments.retainAll(segmentsToProcess);
+
+                  excludedKeys = segments.stream().flatMap(s -> results.referenceArray.get(s).stream())
+                          .collect(Collectors.toSet());
+               } else {
+                  segments = null;
+                  excludedKeys = Collections.emptySet();
+               }
+               KeyTrackingTerminalOperation<Object, R, Object> op = iteratorOperation.getOperation(
+                       intermediateOperations, supplierForSegments(ch, segmentsToProcess, excludedKeys, !stayLocal),
+                       distributedBatchSize);
+               if (!stayLocal) {
+                  UUID id = csm.remoteStreamOperationRehashAware(iteratorParallelDistribute, parallel, ch,
+                          segmentsToProcess, keysToFilter, new AtomicReferenceArrayToMap<>(results.referenceArray),
+                          includeLoader, op, results);
+                  supplier.pending = id;
+                  try {
+                     if (runLocal) {
+                        performLocalRehashAwareOperation(results, segmentsToProcess, ch, segments, op,
+                                () -> ch.getPrimarySegmentsForOwner(localAddress), id);
+                     }
+                     try {
+                        if (!csm.awaitCompletion(id, 30, TimeUnit.SECONDS)) {
+                           throw new TimeoutException();
+                        }
+                     } catch (InterruptedException e) {
+                        throw new CacheException(e);
+                     }
+                     segmentsToProcess = segmentsToProcess(supplier, results, segmentsToProcess, id);
+                  } finally {
+                     csm.forgetOperation(id);
+                  }
+               } else {
+                  performLocalRehashAwareOperation(results, segmentsToProcess, ch, segments, op,
+                          () -> ch.getSegmentsForOwner(localAddress), null);
+                  segmentsToProcess = segmentsToProcess(supplier, results, segmentsToProcess, null);
+               }
+            } while (!complete.get());
+         } catch (CacheException e) {
+            log.trace("Encountered local cache exception for stream", e);
+            supplier.close(e);
+         } catch (Throwable t) {
+            log.trace("Encountered local throwable for stream", t);
+            supplier.close(new CacheException(t));
+         }
+      });
+   }
 
-      CloseableIterator<R> closeableIterator = new CloseableSuppliedIterator<>(supplier);
-      onClose(() -> supplier.close());
-      return closeableIterator;
+   private Set<Integer> segmentsToProcess(IteratorSupplier<R> supplier, KeyTrackingConsumer<Object, R> results,
+                                          Set<Integer> segmentsToProcess, UUID id) {
+      String strId = id == null ? "local" : "identifier " + id;
```

Comment:
I would like if possible to avoid the string concat, maybe just leave it as id or some way of using ``if (log.traceEnabled())`` ?